config/routes.rb

```diff
@@ -1,81 +1,32 @@
 # frozen_string_literal: true
 
 # rubocop:disable Metrics/BlockLength
-# TODO: see https://github.com/SUSE/Portus/issues/1469
 Rails.application.routes.draw do
-  resources :errors, only: [:show]
-  resources :teams, only: %i[index show update] do
-    member do
-      get "typeahead/:query" => "teams#typeahead", :defaults => { format: "json" }
-    end
-  end
-  get "/teams/typeahead/:query" => "teams#all_with_query", :defaults => { format: "json" }
 
+	def draw(routes_module)
+    instance_eval(File.read(Rails.root.join("config/routes/#{routes_module}.rb")))
+	end
+  resources :errors, only: [:show]
+	draw :teams
   resources :help, only: [:index]
 
   resources :team_users, only: %i[create destroy update]
-  resources :namespaces, only: %i[index show update] do
-    put "change_visibility", on: :member
-    resources :webhooks do
-      resources :headers, only: %i[create destroy], controller: :webhook_headers
-      resources :deliveries, only: [:update], controller: :webhook_deliveries
-      member do
-        put "toggle_enabled"
-      end
-    end
-  end
-  get "namespaces/typeahead/:query" => "namespaces#typeahead", :defaults => { format: "json" }
-
-  resources :repositories, only: %i[index show destroy] do
-    post :toggle_star, on: :member
-    resources :comments, only: %i[create destroy]
-  end
-
+	draw :namespaces
+	draw :repositories
   resources :tags, only: %i[show destroy]
-
   resources :application_tokens, only: %i[create destroy]
-
-  devise_for :users, controllers: { registrations:      "auth/registrations",
-                                    sessions:           "auth/sessions",
-                                    passwords:          "passwords",
-                                    omniauth_callbacks: "auth/omniauth_callbacks" }
+	draw :user
   resource :dashboard, only: [:index]
   resources :search, only: [:index]
   resources :explore, only: [:index]
-
-  authenticated :user do
-    root "dashboard#index", as: :authenticated_root
-  end
-
-  devise_scope :user do
-    root to: "auth/sessions#new"
-    put "toggle_enabled/:id", to: "auth/registrations#toggle_enabled", as: :toggle_enabled
-  end
-
-  namespace :v2, module: "api/v2", defaults: { format: :json } do
-    root to: "ping#ping", as: :ping
-    resource :token, only: [:show]
-    resource :webhooks, only: [] do
-      resources :events, only: [:create]
-    end
-  end
-
+	draw :v2
```

Comment:
Rename this to `registries` or something like that, since `v2` can be troublesome...

Reply:
done.

Reply:
Is `registries` the best option we can think of? We have some admin registries thing and might have more in the future, I guess renaming it to `registries` would make things more confusing.

Reply:
Yeah ... What about `events` ? Since it's all about events getting in and out.

Reply:
`ping`, `token`, `webhooks/events`... 🤔  

Btw, do we still need `ping`? We now have a `health` endpoint with the grape api. Without `ping` there are 2 left... They belong to `api/v2` and are used as api endpoints, I still think `legacy_api` could be a good call but I dunno.

Reply:
Bear in mind that those are endpoints dealing with registry events, and ping is an action that can be performed by the registry to Portus iirc 😉 The `token` endpoint is the one responding to JWT requests from the registry.

We could move these endpoints to the portus API, now that we have one, but then you'd force people to change the config of the registry from v2 to v1 (v2 of what we had because we were stating v2 of the registry, and then v1 for the portus API). One way to alleviate this is to move these endpoints when we create the v2 of the portus API. Then this move would be transparent to administrators. What do you think ?

Reply:
> Bear in mind that those are endpoints dealing with registry events, and ping is an action that can be performed by the registry to Portus iirc 😉 The token endpoint is the one responding to JWT requests from the registry.

Okie.

>  We could move these endpoints to the portus API, now that we have one, but then you'd force people to change the config of the registry from v2 to v1 (v2 of what we had because we were stating v2 of the registry, and then v1 for the portus API). One way to alleviate this is to move these endpoints when we create the v2 of the portus API. Then this move would be transparent to administrators. What do you think ?

Agreed! 👍 

Maybe `registry_events`, `registry_api`, ...?

Reply:
Sounds good to me 👍

Reply:
I just added this point in #1500.